Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -1557,6 +1557,10 @@ - (void)loadAnotherPageOfMessages
     //
     // Otherwise, tapping on "load more messages" autoscrolls you downward which is completely wrong.
     if (hasEarlierUnseenMessages) {
+        // Ensure view items are updated before trying to scroll to the
+        // unread indicator.
+        [[OWSPrimaryStorage sharedManager] updateUIDatabaseConnectionToLatest];
```

Comment:
This should be pretty rare, but I realized that we missed an edge case yesterday.

Reply:
If you are going to add a comment, try to have it explain "why", not "what".  The code is already as clear as the comment, and after reading I still don't understand why we want to update the uiDB outside of the normal yapDBModified cycle.

The one scenario I know of when we need to update the uiDB outside of the normal yapDBModified cycle is immediately after the `ensureDynamicInteractions` is called. We need it in that one place because we are performing a synchronous write transaction on the main thread, and want the side effects of that transaction (e.g. a newly created unread indicator) to be available to the rendering process. Waiting for yapDBModified would be too late because it happens async.

As far as I can tell, we're opening no write transaction here, so none of that applies. To be clear, I don't think this will do any significant harm, but I want to make sure we're not superstitiously sprinkling this fix around where it doesn't apply.

Reply:
> We need it in that one place because we are performing a synchronous write transaction on 

A pretty similar scenario is happening here.

> loadNMoreMessages calls resetMappings which calls ensureDynamicInteractions,
> which may move the unread indicator, and for scrollToUnreadIndicatorAnimated
> to work properly, the view items need to be updated to reflect that change.

I've updated the comment to reflect that.